### PR TITLE
[Fixes] Virtual Machine - fixed the subnetId (should be subnetResourceId) old reference for the…

### LINKED ISSUE #2114: [Bug Report]: wrong parameter name in the readme.md of the Microsoft.Compute/virtualMachines
### Describe the bug

wrong parameter name in the readme.md of the Microsoft.Compute/virtualMachines

### To reproduce

subnetID: should be changed to subnetResourceId

### Code snippet

```Bicep
"nicConfigurations": {
  "value": [
    {
      "nicSuffix": "-nic-01",
      "deleteOption": "Delete", // Optional. Can be 'Delete' or 'Detach'
      "ipConfigurations": [
        {
          "name": "ipconfig1",
          "subnetId": "/subscriptions/<subscriptionId>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vNetName>/subnets/<subnetName>",
          "pipConfiguration": {
            "publicIpNameSuffix": "-pip-01",
            "roleAssignments": [
              {
                "roleDefinitionIdOrName": "Reader",
                "principalIds": [
                  "<principalId>"
                ]
              }
            ]
          }
```

### Relevant log output

```Shell
az deployment sub create -c --location westeurope --template-file deploy.bicep --parameters @(".test/parameters.json")          in pwsh at 16:02:16 
C:\Users\olivergulich\GitHub\CARML-ChrisS\ResourceModules\modules\Microsoft.Compute\virtualMachines\deploy.bicep(446,9) : Warning use-resource-id-functions: If property "id" represents a resource ID, it must use a symbolic resource reference, be a parameter or start with one of these functions: extensionResourceId, guid, if, reference, resourceId, subscription, subscriptionResourceId, tenantResourceId. [https://aka.ms/bicep/linter/use-resource-id-functions]
```

### PR DESCRIPTION
**Pull request by oliverlabs**
… nicConfigurations array in the readme.md

Fixes **Bug Issue #2114**

Resolves #2114

# Description

>Thank you for your contribution !

> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [x] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Comment by oliverlabs** [comment]
Thanks, @eriqua

File: modules/Microsoft.Compute/virtualMachines/readme.md [hunk 1]
```diff
@@ -13,105 +13,104 @@ This module deploys one Virtual Machine with one or multiple nics and optionally
...
-| `Microsoft.Automanage/configurationProfileAssignments` | [2021-04-30-preview](https://docs.microsoft.com/en-us/azure/templates) |
-| `Microsoft.Compute/virtualMachines` | [2021-07-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Compute/2021-07-01/virtualMachines) |
-| `Microsoft.Compute/virtualMachines/extensions` | [2021-07-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Compute/2021-07-01/virtualMachines/extensions) |
-| `Microsoft.Insights/diagnosticSettings` | [2021-05-01-preview](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Insights/2021-05-01-preview/diagnosticSettings) |
-| `Microsoft.Network/networkInterfaces` | [2021-08-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Network/2021-08-01/networkInterfaces) |
-| `Microsoft.Network/publicIPAddresses` | [2021-08-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Network/2021-08-01/publicIPAddresses) |
+| Resource Type                                                                         | API Version                                                                                                                                                   |
```
Comment (MariusStorhaug) [inline modules/Microsoft.Compute/virtualMachines/readme.md:16]: Could you regenerate this file using the [ReadMe generator](https://github.com/Azure/ResourceModules/wiki/Contribution%20guide%20-%20Generate%20module%20Readme)? Seems there are some extensions to VSCode which is formatting your tables, which we have disabled with the .vscode settings.

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]
Some changes suggested:
- Clean up the settings in .vscode/settings.json, unless there is a valid justification to keep the setting.
- Regenerate the readme using the provided tool.

File: .vscode/settings.json [hunk 2]
```diff
@@ -50,5 +50,5 @@
   "spellright.language": [
     "en"
   ],
-  "yaml.format.singleQuote": true
+  "yaml.format.singleQuote": true,
```
Comment (MariusStorhaug) [inline .vscode/settings.json:53]:
```suggestion
  "yaml.format.singleQuote": true
```

**Review by MariusStorhaug — CHANGES_REQUESTED** [review]

File: modules/Microsoft.Compute/virtualMachines/readme.md (hunk 1 above)
Comment (oliverlabs) [reply modules/Microsoft.Compute/virtualMachines/readme.md]: Hi @MariusStorhaug. Thanks. This was not intentional. It is caused by Windows converting LF to CRLF at the end of lines. I have fixed this in the latest commit. Please check and let me know if you're happy.

**Review by MariusStorhaug — APPROVED** [review]

File: modules/Microsoft.Compute/virtualMachines/readme.md (hunk 1 above)
Comment (MariusStorhaug) [reply modules/Microsoft.Compute/virtualMachines/readme.md]: Thank you for the work and contribution! :D Fixed the last thing, just a comma.

**Pull request merged**